src/ucp/wireup/wireup_ep.c

```diff
@@ -489,7 +489,8 @@ ssize_t ucp_wireup_ep_sockaddr_fill_private_data(void *arg, const char *dev_name
     uint64_t tl_bitmap;
     char aux_tls_str[64];
 
-    status = ucp_worker_get_address(worker, &worker_address, &address_length);
+    status = ucp_address_pack(worker, NULL, -1, NULL, &address_length,
```

Comment:
why is this removing an 'extra worker lock'? where is the first one? this function is invoked from the client from the async thread without grabbing the lock.

Reply:
Async thread (rdmacm callback) blocks async context then tries to grabs the worker's lock (missed but needed for EP creation), meanwhile, main thread grabs the worker's lock doing 1st sending and trying to block async context on wireup path.
There is a race condition without missed lock for EP creation.